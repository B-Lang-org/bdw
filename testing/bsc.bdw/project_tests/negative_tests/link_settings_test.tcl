
# Sets the compilation type to bsc
handlers::set_compilation_type bsc

# Opens the specified project
handlers::open_project project.bspec

# Sets the compilation type to bsc
handlers::set_compilation_type bsc

# Sets the link type without arguments 
handlers::set_link_type 

# Sets the link type with wrong argument
handlers::set_link_type wrong 

# Sets the link bsc type and options without arguments
handlers::set_link_bsc_options

# Sets the link bsc output file, directory and options
handlers::set_link_bsc_options mmm -wrong

# Sets the link bsc output file, directory and options
handlers::set_link_bsc_options mmm -path /wrong

# Sets the link make options
handlers::set_link_make_options Makefile

# Sets the link type to make
handlers::set_link_type make

# Sets the link bsc options
handlers::set_link_bsc_options mm

# Sets the link make options
handlers::set_link_make_options

# Sets the link make options
handlers::set_link_make_options mmm 

# Sets the link make options
handlers::set_link_make_options Makefile -tget ta

# Sets the link make options
handlers::set_link_make_options Makefile -target

# Sets the link type to bsc
handlers::set_link_custom_command puts

# Sets the link type to make
handlers::set_link_type custom_command 

# Sets the link type to bsc
handlers::set_link_custom_command

# Closes the project
handlers::close_project

# Exits the workstation
exit
