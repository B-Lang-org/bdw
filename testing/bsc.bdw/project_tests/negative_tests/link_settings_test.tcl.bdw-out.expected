There is no open project.
The link type is not specified
The link type should be bsc, make, custom_command.
The output for link is not specified
Unknown option: -wrong
The path '/wrong' is not a directory.
The link type is not set to "make"  for the current project.
The link type is not set to "bsc"  for the current project.
The makefile for the link is not specified
The file mmm doesn't exist.
Unknown option: -tget
The link type is not set to "custom_comand"  for the current project.
The custom command for the link is not specified
